tests/Unit/Utilities/Test_Functional.cpp

```diff
@@ -0,0 +1,232 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <array>
+#include <cmath>
+#include <complex>
+#include <functional>
+#include <random>
+
+#include "Utilities/Functional.hpp"
+#include "Utilities/TMPL.hpp"
+#include "tests/Utilities/MakeWithRandomValues.hpp"
+
+using std::imag;
+using std::real;
+
+#define MAKE_UNARY_TEST(STRUCTNAME, FUNC)                         \
+  template <typename T, typename UniformGen, typename RandomDist> \
+  void test_func_eval_##FUNC(UniformGen gen, RandomDist dist) {   \
+    T val = dist(gen);                                            \
+    CHECK(STRUCTNAME(val) == FUNC(val));                          \
+  }
+
+#define MAKE_BINARY_TEST(STRUCTNAME, FUNC)                      \
+  template <typename T1, typename T2, typename UniformGen,      \
+            typename RandomDist1, typename RandomDist2>         \
+  void test_func_eval_##FUNC(UniformGen gen, RandomDist1 dist1, \
+                             RandomDist2 dist2) {               \
+    T1 val1 = dist1(gen);                                       \
+    T2 val2 = dist2(gen);                                       \
+    CHECK(STRUCTNAME(val1, val2) == FUNC(val1, val2));          \
+  }
+
+#define MAKE_BINARY_INPLACE_TEST(FUNCNAME, STRUCTNAME, FUNC)        \
+  template <typename T1, typename T2, typename UniformGen,          \
+            typename RandomDist1, typename RandomDist2>             \
+  void test_func_eval_##FUNCNAME(UniformGen gen, RandomDist1 dist1, \
+                                 RandomDist2 dist2) {               \
+    T1 val1 = dist1(gen);                                           \
+    T2 val2 = dist2(gen);                                           \
+    auto result = FUNC(val1, val2);                                 \
+    CHECK(STRUCTNAME(val1, val2) == result);                        \
+    CHECK(val1 == result);                                          \
+  }
+
+MAKE_UNARY_TEST(Abs<>{}, abs)
+MAKE_UNARY_TEST(ACos<>{}, acos)
+MAKE_UNARY_TEST(ACosh<>{}, acosh)
+MAKE_UNARY_TEST(ASin<>{}, asin)
+MAKE_UNARY_TEST(ASinh<>{}, asinh)
+MAKE_UNARY_TEST(ATan<>{}, atan)
```

Comment:
`ATan` -> `Atan`, similar to all the others everywhere that do the current thing.